src/components/button-toggle/button-toggle-group/button-toggle-group.test.tsx

```diff
@@ -0,0 +1,424 @@
+      </ButtonToggleGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.queryByText("Label Help")).not.toBeInTheDocument();
+  expect(screen.queryByText("Field Help")).not.toBeInTheDocument();
+});
+
+test("should only allow the first button to be tabbable when none are selected", async () => {
+  const user = userEvent.setup();
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value={undefined}
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  // need await as tabindex conditions were not computed before assertion and test would fail
+  await user.tab();
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveFocus();
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveAttribute(
+    "tabindex",
+    "0"
+  );
+  expect(screen.getByRole("button", { name: "Bar" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+});
+
+test("should only allow a selected button to be tabbable", () => {
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value="bar"
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+      <ButtonToggle value="baz">Baz</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+  expect(screen.getByRole("button", { name: "Bar" })).toHaveAttribute(
+    "tabindex",
+    "0"
+  );
+  expect(screen.getByRole("button", { name: "Baz" })).toHaveAttribute(
+    "tabindex",
+    "-1"
+  );
+});
+
+test("should focus on the previous button when the left arrow key is pressed", async () => {
+  const user = userEvent.setup();
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value="bar"
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+      <ButtonToggle value="baz">Baz</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  await user.tab();
+  await user.keyboard("{arrowleft}");
+  expect(screen.getByRole("button", { name: "Foo" })).toHaveFocus();
+});
+
+test("should focus on the last button when the first button is focused and the left arrow key is pressed", async () => {
+  const user = userEvent.setup();
+  render(
+    <ButtonToggleGroup
+      id="button-toggle-group-id"
+      onChange={() => {}}
+      value="foo"
+    >
+      <ButtonToggle value="foo">Foo</ButtonToggle>
+      <ButtonToggle value="bar">Bar</ButtonToggle>
+      <ButtonToggle value="baz">Baz</ButtonToggle>
+    </ButtonToggleGroup>
+  );
+
+  await user.tab();
+  await user.keyboard("{arrowleft}");
+  expect(screen.getByRole("button", { name: "Baz" })).toHaveFocus();
+});
+
+test("should focus on the next button when the right arrow key is pressed", async () => {
```

Comment:
**suggestion**: similarly with this test just to make the precondition a bit clearer:


```suggestion
test("focuses the button right of the currently selected button, when the right arrow key is pressed", async () => {
```

Reply:
same as above for me